exporter/awsemfexporter/grouped_metric_test.go

```diff
@@ -202,26 +198,39 @@ func TestAddToGroupedMetric(t *testing.T) {
 		assert.Equal(t, 9, metrics.Len())
 
 		for i := 0; i < metrics.Len(); i++ {
-			err := addToGroupedMetric(metrics.At(i), groupedMetrics, metadata, true, logger, nil, nil)
+			err := addToGroupedMetric(metrics.At(i), groupedMetrics, generateTestMetricMetadata(namespace, timestamp, logGroup, logStreamName, instrumentationLibName, metrics.At(i).Type()), true, logger, nil, nil)
 			assert.Nil(t, err)
 		}
 
-		assert.Equal(t, 1, len(groupedMetrics))
+		assert.Equal(t, 4, len(groupedMetrics))
 		for _, group := range groupedMetrics {
-			assert.Equal(t, 6, len(group.metrics))
 			for metricName, metricInfo := range group.metrics {
-				if metricName == "double-histogram" || metricName == "summary" {
-					assert.Equal(t, "Seconds", metricInfo.unit)
-				} else {
+				switch metricName {
+				case "int-gauge", "double-gauge":
+					assert.Equal(t, 2, len(group.metrics))
 					assert.Equal(t, "Count", metricInfo.unit)
+					assert.Equal(t, generateTestMetricMetadata(namespace, timestamp, logGroup, logStreamName, instrumentationLibName, pmetric.MetricTypeGauge), group.metadata)
+				case "int-sum", "double-sum":
+					assert.Equal(t, 2, len(group.metrics))
+					assert.Equal(t, "Count", metricInfo.unit)
+					assert.Equal(t, generateTestMetricMetadata(namespace, timestamp, logGroup, logStreamName, instrumentationLibName, pmetric.MetricTypeSum), group.metadata)
+				case "double-histogram":
+					assert.Equal(t, 1, len(group.metrics))
+					assert.Equal(t, "Seconds", metricInfo.unit)
+					assert.Equal(t, generateTestMetricMetadata(namespace, timestamp, logGroup, logStreamName, instrumentationLibName, pmetric.MetricTypeHistogram), group.metadata)
+				case "summary":
+					assert.Equal(t, 1, len(group.metrics))
+					assert.Equal(t, "Seconds", metricInfo.unit)
+					assert.Equal(t, generateTestMetricMetadata(namespace, timestamp, logGroup, logStreamName, instrumentationLibName, pmetric.MetricTypeSummary), group.metadata)
+				default:
+					assert.FailNow(t, "Unhandled metric not expected")
```

Comment:
**Nit** Would be good to show metric type too

Reply:
Done